File: Hash3.java
// 해쉬 (위장)문제
import java.util.*;

class Hash3 {
    public int solution(String[][] clothes) {
        int answer = 1;
        
        HashMap<String,Integer> clothesMap = new HashMap<String,Integer>();
        for(int i=0;i<clothes.length;i++){
            String cls = clothes[i][1];
            boolean check = clothesMap.containsKey(cls);
            if(check){
                int count = clothesMap.get(cls);
                clothesMap.remove(cls);
                clothesMap.put(cls,count+1);
            }else{
                clothesMap.put(cls,1);
            }
        }
        for(String s: clothesMap.keySet()){
            int tmp = clothesMap.get(s);
            answer = answer * (tmp+1);
        }
        
        answer = answer -1;
        
        return answer;
    }
}

File: Hash3.js
function Hash3(clothes) {
    let answer = 1;
    let map = new Map();
    for(let i=0;i<clothes.length;i++){
        let cls = clothes[i][1];
        if(map.get(cls)!=undefined){
            let count = map.get(cls);
            map.delete(cls);
            map.set(cls,count+1);
        }
        else{
            map.set(cls,1);
        }
    }
    for(let c of map.values()){
        answer *= (c+1);
    }
    
    answer -= 1;
    
    return answer;
}
